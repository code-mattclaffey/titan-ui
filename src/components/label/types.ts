import { GenericComponent } from '../../module';

export interface LabelInterface extends GenericComponent {
    htmlFor: string;
    labelText?: string;
    required?: boolean;
}
