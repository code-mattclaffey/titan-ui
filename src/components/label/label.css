/*
    Anchor
*/

:root {
    --label-color: var(--base-font-color);
    --label-font-weight: var(--base-font-weight-semibold);
    --label-font-weight-alt: var(--base-font-weight-normal);
    --label-required-content: ' *';
    --label-required-color: var(--warning-color);
}

.e-label {
    color: var(--label-color);
    font-weight: var(--label-font-weight);
}

.e-label--required .e-label__text::after {
    content: var(--label-required-content);
    color: var(--label-required-color);
}

.e-checkbox__input ~ .e-label__text {
    font-weight: var(--label-font-weight-alt);
}

.e-checkbox__input[disabled] ~ .e-label__text {
    opacity: 0.8;
}
