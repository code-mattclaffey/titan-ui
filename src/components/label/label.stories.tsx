import React from 'react';

import { Label } from '.';

export default {
    title: 'Label',
    componentSubtitle: '',
};

export const label = () => (
    <Label htmlFor="test-input" labelText="Test Label" />
);
