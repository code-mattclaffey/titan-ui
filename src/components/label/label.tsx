import React from 'react';
import classNames from 'classnames';
import { LabelInterface } from './types';

export const Label = ({
    additionalClassNames,
    children,
    htmlFor,
    labelText,
    required = true,
    ...rest
}: LabelInterface) => {
    const classes = classNames('e-label', additionalClassNames, {
        'e-label--required': required,
    });

    return (
        <label
            className={classes}
            htmlFor={htmlFor}
            {...rest}
            data-testid="e-label"
        >
            {labelText && (
                <span className="e-label__text">{labelText}</span>
            )}
            {children}
        </label>
    );
};
