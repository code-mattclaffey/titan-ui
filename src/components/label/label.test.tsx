import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Label } from './label';

afterEach(() => {
    cleanup();
});

describe('<Label />', () => {
    it('should match the snapshot', () => {
        const { container } = render(
            <>
                <Label
                    htmlFor="test"
                    labelText="Hello World"
                    required={false}
                />
                ,
                <input id="test" />
            </>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should have required * when the prop is set', () => {
        const { queryByTestId } = render(
            <>
                <Label
                    htmlFor="test"
                    labelText="Hello World"
                    required
                />
                ,
                <input id="test" />
            </>,
        );

        expect(
            queryByTestId('e-label').classList.contains(
                'e-label--required',
            ),
        ).toBeTruthy();
    });
});
