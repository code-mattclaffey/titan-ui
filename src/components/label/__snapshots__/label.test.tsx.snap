// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match the snapshot 1`] = `
<div>
  <label
    class="e-label"
    data-testid="e-label"
    for="test"
  >
    <span
      class="e-label__text"
    >
      Hello World
    </span>
  </label>
  ,
  <input
    id="test"
  />
</div>
`;
